### DISCUSSION TITLE
How to get the current theme ?

### DISCUSSION THREAD

**Discussion by waleedsalah4** [fuma-nama/fumadocs#1140] [Q&A] [answered]
I need to extract the current theme to update some content based on the current one? Is there a hook or somewhere I can get the current theme from ??

**Comment by fuma-nama** [comment]
install and use `next-themes`

**Reply by waleedsalah4** [reply]
```tsx
"use client";

import * as React from "react";
import { ThemeProvider as NextThemesProvider } from "next-themes";

export function ThemeProvider({
  children,
  ...props
}: React.ComponentProps<typeof NextThemesProvider>) {
  return <NextThemesProvider {...props}>{children}</NextThemesProvider>;
}
```

```tsx
export default function Layout({ children }: { children: ReactNode }) {
  return (
    <html lang="en" className={inter.className} suppressHydrationWarning>
      <body>
        <ThemeProvider>
          <RootProvider>{children}</RootProvider>
        </ThemeProvider>
      </body>
    </html>
  );
}
```

```tsx
"use client";
import Image from "next/image";
import { useTheme } from "next-themes";

const Logo = () => {
  const { theme } = useTheme();
  console.log("theme: ", theme);

  return (
    <Image
      src={theme === "light" ? "/black-logo.svg" : "/white-logo.svg"}
      alt="logo"
      width={100}
      height={50}
      className="pl-2"
    />
  );
};
export default Logo;
```

Using this code and getting the current theme,
The `theme` always gives me the initial value and when ever I change the current theme from the switcher in the header or side bar doesn't get the correct theme value, as I said it always gives me the initial value

**Reply by fuma-nama** [reply]
You don't need another provider, and you can use `resolvedTheme`, see their docs for details

**Reply by waleedsalah4** [reply]
Removing the provider from theme provider 
Makes both `theme` and `resolvedTheme` to be `undefined` 
And never changes even the theme changes 
![IMG-20241208-WA0000](https://github.com/user-attachments/assets/383833a3-c259-4f75-8871-330858b32d28)

here's the code
https://stackblitz.com/~/github.com/waleedsalah4/docs

this logo component is used inside the top navbar

**Reply by fuma-nama** [reply, marked as answer]
You somehow installed Fumadocs v13, which uses `next-themes` 0.3.0 (an older version), you have to make sure you have installed the exact version of `next-themes` Fumadocs uses, or upgrade to latest versions of Fumadocs to avoid this issue

**Reply by waleedsalah4** [reply]
Thanks, it works after matching the versions

**Reply by amin-bake** [reply]
Hello @fuma-nama I think I might be having a similar issue but with the latest versions where both `theme` and `resolvedTheme` return `undefined`. 
**Versions**
![version mismatch](https://github.com/user-attachments/assets/a0464670-41f3-433d-9874-3a25cc18aa87)
> Please provide Help!

**Reply by grmkris** [reply]
same here

**Reply by fuma-nama** [reply]
The latest `fumadocs-ui` is `14.7.1`, and try pnpm if npm doesn't work, could be a problem of package manager

**Reply by fuma-nama** [reply]
Or you can replace the themes provider with your own from `next-themes`, e.g. in `app/layout.tsx`:

```tsx
<ThemeProvider
        attribute="class"
        defaultTheme="system"
        enableSystem
        disableTransitionOnChange>
      ...
</ThemeProvider>
```

And disable the default one:

```tsx
<RootProvider theme={{ enabled: false }}>...</RootProvider>
```

**Reply by grmkris** [reply]
yes that's what i did in my case.. replacing theme provider

**Reply by amin-bake** [reply]
@fuma-nama the default `<RootProvider/>` actually worked. I realized my mistake.

**Reply by rockywangxiaolei** [reply]
@fuma-nama i am new to fumadocs, and would like to ask if i am using latest version, how can i get the current theme? do i still need to install next-themes as described above?
Thanks.
=================
 "next": "15.5.4",
    "react": "^19.1.1",
    "react-dom": "^19.1.1",
    "fumadocs-ui": "15.8.0",
    "fumadocs-core": "15.8.0",
    "fumadocs-mdx": "12.0.1"

**Reply by fuma-nama** [reply]
yes

**Comment by rockywangxiaolei** [comment]
Sorry but i used the latest fumadocs and tried to implement the same features like @waleedsalah4 , but it comes to error, @fuma-nama 

**Versions used:**
 "dependencies": {
    "fumadocs-core": "16.1.0",
    "fumadocs-mdx": "14.0.3",
    "fumadocs-ui": "16.1.0",
    "lucide-react": "^0.552.0",
    "next": "16.0.1",
    "next-themes": "^0.4.6",
    "react": "^19.2.0",
    "react-dom": "^19.2.0"
  },


**Code:**

import type { BaseLayoutProps } from 'fumadocs-ui/layouts/shared';
import Logo from "@/components/logo";

export function baseOptions(): BaseLayoutProps {
  return {
    nav: {
      title:  (
        <>
          <Logo/>
          Docs
        </>
      ),
    },
  };
}

"use client";
import Image from "next/image";
import { useTheme } from "next-themes";

const Logo = () => {
  const { theme, resolvedTheme, setTheme } = useTheme();

  return (
    <Image
      src={theme === "light" ? "/tripo-light.svg" : "/tripo-dark.svg"}
      alt="logo"
      width={100}
      height={50}
      className="pl-2"
    />
  );
};
export default Logo;

**Error:**
A tree hydrated but some attributes of the server rendered HTML didn't match the client properties. This won't be patched up. This can happen if a SSR-ed Client Component used:
- A server/client branch `if (typeof window !== 'undefined')`.
- Variable input such as `Date.now()` or `Math.random()` which changes each time it's called.
- Date formatting in a user's locale which doesn't match the server.
- External changing data without sending a snapshot of it along with the HTML.
- Invalid HTML tag nesting.

**Comment by fuma-nama** [comment]
It's normal, see https://github.com/pacocoursey/next-themes if you're new to React/the library.
